### ISSUE TITLE
Incorrect rule: "E1019: Sub parameter should be an object of 1 or string for..."

### ISSUE THREAD

**Issue by CharlesStover** [aws-cloudformation/cfn-python-lint#1772]
*cfn-lint version: 0.40.0*

I am getting an incorrect error that [`E1019`](https://github.com/aws-cloudformation/cfn-python-lint/blob/master/docs/rules.md#E1019)` Sub parameter should be an object of 1 or string for...` when using YAML:

```
        - Fn::Sub:
            - 'example-${Var}-${Var2}'
            - Var: 123
              Var2: 456
```

Official docs: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-sub.html

Official docs sample:
```
Fn::Sub:
  - String
  - Var1Name: Var1Value
    Var2Name: Var2Value
```

**Comment by PatMyron** [comment]
Haven't looked into this yet, but getting thrown here:
https://github.com/aws-cloudformation/cfn-python-lint/blob/4ef277bb637c533582c2b10d3589b4aa3623f3bf/src/cfnlint/rules/functions/Sub.py#L88-L91

**Comment by kddejong** [comment]
@PatMyron that was the correct spot.  Updated the rule to handle this scenario.
